ext/fiddle/extconf.rb

```diff
@@ -218,35 +218,40 @@ def enable_debug_build_flag(flags)
   $LOCAL_LIBS.prepend("#{libffi.a} ").strip! # to exts.mk
   $INCFLAGS.gsub!(/-I#{libffi.dir}/, '-I$(LIBFFI_DIR)')
 end
-create_makefile 'fiddle' do |conf|
-  if !libffi
-    next conf << "LIBFFI_CLEAN = none\n"
-  elsif $gnumake && !$nmake
-    submake_arg = "-C $(LIBFFI_DIR)\n"
-  else
-    submake_pre = "cd $(LIBFFI_DIR) && #{config_string("exec")}".strip
-  end
-  if $nmake
-    cmd = "$(RUBY) -C $(LIBFFI_DIR) #{libffi_config} --srcdir=$(LIBFFI_SRCDIR)"
-  else
-    cmd = "cd $(LIBFFI_DIR) && #$exec $(LIBFFI_SRCDIR)/configure #{libffi.opt}"
+
+if RUBY_ENGINE == 'ruby'
+  create_makefile 'fiddle' do |conf|
+    if !libffi
+      next conf << "LIBFFI_CLEAN = none\n"
+    elsif $gnumake && !$nmake
+      submake_arg = "-C $(LIBFFI_DIR)\n"
+    else
+      submake_pre = "cd $(LIBFFI_DIR) && #{config_string("exec")}".strip
+    end
+    if $nmake
+      cmd = "$(RUBY) -C $(LIBFFI_DIR) #{libffi_config} --srcdir=$(LIBFFI_SRCDIR)"
+    else
+      cmd = "cd $(LIBFFI_DIR) && #$exec $(LIBFFI_SRCDIR)/configure #{libffi.opt}"
+    end
+    sep = "/"
+    seprpl = config_string('BUILD_FILE_SEPARATOR') {|s| sep = s; ":/=#{s}" if s != "/"} || ""
+    conf << <<-MK.gsub(/^ +| +$/, '')
+     PWD =
+     LIBFFI_CONFIGURE = #{cmd}
+     LIBFFI_ARCH = #{libffi.arch}
+     LIBFFI_SRCDIR = #{libffi.srcdir.sub(libffi.dir, '$(LIBFFI_DIR)')}
+     LIBFFI_DIR = #{libffi.dir}
+     LIBFFI_A = #{libffi.a.sub(libffi.dir, '$(LIBFFI_DIR)')}
+     LIBFFI_CFLAGS = #{libffi.cflags}
+     LIBFFI_LDFLAGS = #{libffi.ldflags}
+     FFI_H = $(LIBFFI_DIR)/include/ffi.h
+     SUBMAKE_PRE = #{submake_pre}
+     SUBMAKE_ARG = #{submake_arg}
+     LIBFFI_CLEAN = libffi
+    MK
   end
-  sep = "/"
-  seprpl = config_string('BUILD_FILE_SEPARATOR') {|s| sep = s; ":/=#{s}" if s != "/"} || ""
-  conf << <<-MK.gsub(/^ +| +$/, '')
-   PWD =
-   LIBFFI_CONFIGURE = #{cmd}
-   LIBFFI_ARCH = #{libffi.arch}
-   LIBFFI_SRCDIR = #{libffi.srcdir.sub(libffi.dir, '$(LIBFFI_DIR)')}
-   LIBFFI_DIR = #{libffi.dir}
-   LIBFFI_A = #{libffi.a.sub(libffi.dir, '$(LIBFFI_DIR)')}
-   LIBFFI_CFLAGS = #{libffi.cflags}
-   LIBFFI_LDFLAGS = #{libffi.ldflags}
-   FFI_H = $(LIBFFI_DIR)/include/ffi.h
-   SUBMAKE_PRE = #{submake_pre}
-   SUBMAKE_ARG = #{submake_arg}
-   LIBFFI_CLEAN = libffi
-  MK
+else
+  File.write('Makefile', dummy_makefile("").join)
```

Comment:
There is a ton of logic above we don't want to do.

So check very early the RUBY_ENGINE, just before `require 'mkmf'` and if no "ruby" then do this + `return`